File: src/dataset/Java/food-chain_exercism-uzilan/gemini/GeminiFlash_FoodChain_v1.java
public class FoodChain {

    private static final String[] ANIMALS = {"fly", "spider", "bird", "cat", "dog", "goat", "cow", "horse"};
    private static final String[] DESCRIPTIONS = {
            "I don't know why she swallowed the fly. Perhaps she'll die.",
            "It wriggled and jiggled and tickled inside her.",
            "How absurd to swallow a bird!",
            "Imagine that, to swallow a cat!",
            "What a hog, to swallow a dog!",
            "Just opened her throat and swallowed a goat!",
            "I don't know how she swallowed a cow!",
            "She's dead, of course!"
    };

    public String verse(int verse) {
        int index = verse - 1;
        StringBuilder sb = new StringBuilder();
        sb.append("I know an old lady who swallowed a ").append(ANIMALS[index]).append(".\n");
        sb.append(DESCRIPTIONS[index]).append("\n");

        if (verse < ANIMALS.length) {
            for (int i = verse - 1; i > 0; i--) {
                sb.append("She swallowed the ").append(ANIMALS[i]).append(" to catch the ").append(ANIMALS[i - 1]);
                if (i == 1) {
                    sb.append(" that wriggled and jiggled and tickled inside her.\n");
                } else {
                    sb.append(".\n");
                }
            }
            sb.append("I don't know why she swallowed the fly. Perhaps she'll die.\n");
        }

        return sb.toString().trim();
    }

    public String verses(int startVerse, int endVerse) {
        StringBuilder sb = new StringBuilder();
        for (int i = startVerse; i <= endVerse; i++) {
            sb.append(verse(i));
            if (i < endVerse) {
                sb.append("\n\n");
            }
        }
        return sb.toString();
    }
}

File: src/dataset/python/food-chain_exercism-fruit-in/gemini/GeminiFlash_food_chain_v1.py
animals = ["fly", "spider", "bird", "cat", "dog", "goat", "cow", "horse"]
phrases = [
    "It wriggled and jiggled and tickled inside her.",
    "How absurd to swallow a bird!",
    "Imagine that, to swallow a cat!",
    "What a hog, to swallow a dog!",
    "Just opened her throat and swallowed a goat!",
    "I don't know how she swallowed a cow!",
    "She's dead, of course!",
]

spider_phrase = " that" + phrases[0][2:]
fly_phrase = "I don't know why she swallowed the fly. Perhaps she'll die."


def verse(n):
    ret = ["I know an old lady who swallowed a {}.".format(animals[n - 1])]
    if n > 1:
        ret.append(phrases[n - 2])
    if n < 8:
        for i in range(n - 1):
            swallowed = animals[n - i - 1]
            to_catch = animals[n - i - 2]
            phrase = "She swallowed the {} to catch the {}.".format(swallowed, to_catch)
            if swallowed == "spider":
                phrase = phrase[:-1] + spider_phrase
            ret.append(phrase)
        ret.append(fly_phrase)

    return ret


def recite(start_verse, end_verse):
    ret = []
    for n in range(start_verse, end_verse + 1):
        ret.extend(verse(n))
        ret.append("")

    return ret[:-1]